[DOXIA-616]

bertysentry opened the pull request:
* Fixed integration tests for **doxia-formats* wrt fenced code blocks
* Upgraded to Doxia 1.10-SNAPSHOT

bertysentry commented:
@michael-o Please note:
* I upgraded to Doxia 1.10-SNAPSHOT (not sure you wanted this here, just let me know)
* I wondered whether we should actually remove these Doxia markdown integration tests from **maven-site-plugin**, as we now have proper integration tests in **doxia-module-markdown** itself. This would prevent such situations where updates in one project (Doxia) requires to update the tests in another project (maven-site-plugin).

Please advise! :-)

michael-o commented:
Thanks, will look into it.

elharo reviewed (commented):
Can this change wait until doxia 1.10 is released?

michael-o commented:
I won't merge before anyway because I never depend on snapshots.

bertysentry commented:
Okay, I will update the PR once Doxia 1.10 is released.

Do you guys have any opinion on the removal of markdown-specific integration tests (especially Doxia-specific) from maven-site-plugin, and reincorporate them in Doxia? This would avoid the chicken-egg situation like this.

michael-o commented:
In my opinion they need to bemoved because we test the entire site generation, and not Doxia specific modules. @hboutemy WDYT?

elharo commented:
I think I agree that these should be moved. If they're only testing doxia functionality, and we can test them in doxia, then that's better. Or perhaps tests belong in both places. Doxia should be an implementation detail of maven-site-plugin. Any tests we have for HTML generation in sites should be independent of the HTML generator we're using.

hboutemy commented:
is it causing issues, so they *need* to be removed?
if you're sure you have sufficient tests nowadays in Doxia, Doxia Sitetools and skins to check that a new feature in a Doxia module has the expected impact when used in a real site, you can drop this end to end test
but I feel that a few end to end tests remain useful to check that the full rendering bring expected result after execution of https://maven.apache.org/plugins/maven-site-plugin/history.html

bertysentry commented:
@hboutemy We will keep some integration tests in *maven-site-plugin* that involve *doxia-module-markdown*. I want to remove specific Doxia tests (like "quotes were mistakenly removed", or "code block must be surrounded with `<div class="source">`.

These can be properly tested in Doxia with unit tests, and also with integration tests. Where applicable, we use integration tests to invoke *maven-site-plugin* with the current version of Doxia and make sure the end-to-end chain works as expected.

I will submit a PR on Doxia to incorporate these specific tests (from *maven-site-plugin*) to *doxia-module-markdown*.

bertysentry commented:
Submitted PR for Doxia: https://github.com/apache/maven-doxia/pull/52

Thank you!

bertysentry commented:
Okay, PR has been updated as below:
* Removed Doxia-specific integration tests
* Added integration tests on the various document formats
* Integration tests on macros are done on the SNIPPET macro from a common Java source file

Please note that the builds on macOS with JDK11, 14 and 15 fail, but these failures are not related to this PR (**master** is failing the same way).

@hboutemy @elharo @michael-o Please review these changes. Thank you!

bertysentry commented:
Oh, last update: I reverted Doxia version to **1.9.1**. MSITE integration tests will pass with both 1.9.1 and 1.10 (no more checks on specific Doxia formatting features).

bertysentry commented:
More build failures: Connection timeout with www.w3.org, which is totally unrelated to this PR.

elharo commented:
We really should resolve the CI issues before proceeding with this or any other changes.

bertysentry commented:
@elharo IMO there is no need to wait to merge this PR. This PR ensures the integration tests won't break when upgrading to Doxia 1.10.

The integration tests that currently fail are only for macOS width JDK 11, 14 and 15:
* Error:  MSITE-665/pom.xml: See [MSITE-665](https://issues.apache.org/jira/browse/MSITE-665) from 2013
* Error:  MSITE-512/pom.xml
* Error:  MSITE-506/pom.xml
* Error:  MSITE-497/pom.xml
* Error:  MSITE-484/pom.xml

They are all ancient issues and completely unrelated to the changes in the integration tests to support Doxia 1.10.

Thanks.

bertysentry commented:
@elharo Any idea on how to retrieve the **build.log** files from GitHub Actions/Workflows? I must be missing something here, but I can't find any link to any downloadable artifact in GitHub Actions... 😅

bertysentry commented:
@hboutemy @michael-o @elharo Will you have a chance to review these changes? This ensures the **maven-site-plugin** integration tests don't break when upgrading from Doxia 1.9.1 to 1.10.

elharo commented:
Before this can move further, someone needs to debug and fix the CI failures. That's likely not caused by this PR, but checking in when the CI is broken is never a good idea.

bertysentry commented:
@elharo We cannot access the integration test logs, how are we going to fix this?

elharo commented:
It might need someone who has more permissions on the project. I'm not sure.

bertysentry commented:
I have full permissions on my fork, and it's running all GitHub Actions as well, and failing at the same place (macOS with JDK11, 14 and 15). But I can't find a way to access the files produced by the build (not even the plugin artifact).

elharo commented:
Some though not all of the errors seem related to this:

: |        [ERROR] Failed to execute goal org.apache.maven.plugins:maven-site-plugin:3.10.0-SNAPSHOT:site (default-cli) on project test: Error parsing '/home/runner/work/maven-site-plugin/maven-site-plugin/target/it/validate/src/site/fml/faq.fml': line [-1] Error validating the model: Connect to www.w3.org:80 [www.w3.org/128.30.52.100] failed: Connection timed out -> [Help 1]

It might be worth understanding why the test is trying to connect to www.3.org. I'm not sure it needs to do that.

bertysentry commented:
This was just a temporary network issue of GitHub Actions. If you re-run the workflow, you won't get these errors.

bertysentry commented:
Also, PR #35 just got merged to *master* 3 days ago, with the same CI failures.

elharo commented:
That one wasn't merged by me. :-)

bertysentry commented:
One recurring error is:
```log
Error: |        [ERROR] Failed to execute goal org.apache.maven.plugins:maven-site-plugin:3.10.0-SNAPSHOT:site (default-cli) on project test: Error parsing '/home/runner/work/maven-site-plugin/maven-site-plugin/target/it/validate/src/site/fml/faq.fml': line [-1] Error validating the model: Connect to www.w3.org:80 [www.w3.org/128.30.52.100] failed: Connection timed out (Connection timed out) -> [Help 1]
```

It happens when the **site:validate** goal is invoked in the integration tests (in **src/it/validate/pom.xml**). Apparently, **site:validate** tries to download the specified XML schema to validate the source documents (FML and XDOC). On GitHub Actions, this appears to fail from time to time.

This is nothing too serious, IMO.

bertysentry commented:
@michael-o @hboutemy @elharo Friendly reminder that this PR is still waiting, and its content is not going to change with regards to the failed builds, which are caused by something else (*master* is failing the same builds the same way).

The more we wait, the more risk we have to get conflicting changes.

Thanks!

michael-o commented:
I won't be able to review before mid Feb due to circumstances here.

slachiewicz commented:
I'll look at it tomorrow

bertysentry commented:
Guys? I'm still waiting for this to be merged so I can move forward with more changes in Doxia. Your help is much appreciated 😉

michael-o commented:
> 
> 
> Guys? I'm still waiting for this to be merged so I can move forward with more changes in Doxia. Your help is much appreciated 😉

As promised, I will pick up as soon as I can, but now Oracle quartely updates are waiting for me...

slachiewicz reviewed (approved):
I squashed commits - looks good to me.
Only MacOS tests failed.
Our Jenkins is green https://ci-builds.apache.org/job/Maven/job/maven-box/job/maven-site-plugin/job/feature%252FDOXIA-616/

michael-o commented:
Let me go through too...

michael-o commented:
> 
> 
> I squashed commits - looks good to me.
> Only MacOS tests failed https://github.com/apache/maven-site-plugin/actions/runs/516170331
> Our Jenkins is green https://ci-builds.apache.org/job/Maven/job/maven-box/job/maven-site-plugin/job/feature%252FDOXIA-616/

I was not able to reproduce this problem multiple times.

bertysentry commented:
Awesome, thank you guys! 😊
